cmd/snap-repair/runner.go

```diff
@@ -80,6 +80,24 @@ func (r *Repair) SetStatus(status RepairStatus) {
 	r.run.SaveState()
 }
 
+// makeRepairSymlink ensures $dir/repair exists and is a symlink to
+// /usr/lib/snapd/snap-repair
+func makeRepairSymlink(dir string) (err error) {
+	// make "repair" binary available to the repair scripts via symlink
+	// to the real snap-repair
+	if err = os.MkdirAll(dir, 0755); err != nil {
+		return err
+	}
+
+	old := filepath.Join(dirs.CoreLibExecDir, "snap-repair")
+	new := filepath.Join(dir, "repair")
+	if osutil.FileExists(new) {
+		return nil
+	}
+
+	return os.Symlink(old, new)
```

Comment:
we need to ignore the error if new already exists

Reply:
Thanks, good catch, there is a os.FileExists() earlier but that is of course bad because of TOUTOC, I pushed a new version that does it properly.